Added CSS to registration page

matthewrockwood opened the pull request:
Added a expense table.
Created it so when you login there is a User object that is the current person logged on. It also fetches their ID from the database to use. Made it so there is only one instance of of userService across the entire project. This is because every controller would be creating a new instance of a userService instead of using the same one. This is apparently a Singleton Pattern. To use it do “private UserService userService = UserService.getInstance();” in any class. Added the add expense method and the delete expense method to and from the database The add expense is utilized in the expense controller, but still needs work. It uses the global userService instance to add items to the expense table and is stored based of the currentUser ID The removeExpense takes in the expensesID to remove. It only works if the user was the one to add it to the table Added a test method to print all the expenses in the expense table

matthewrockwood committed:
Added CSS to registration page
Added a expense table.
Created it so when you login there is a User object that is the current person logged on. It also fetches their ID from the database to use.
Made it so there is only one instance of of userService across the entire project. This is because every controller would be creating a new instance of a userService instead of using the same one. This is apparently a Singleton Pattern. To use it do “private UserService userService = UserService.getInstance();” in any class.
Added the add expense method and the delete expense method to and from the database
The add expense is utilized in the expense controller, but still needs work. It uses the global userService instance to add items to the expense table and is stored based of the currentUser ID
The removeExpense takes in the expensesID to remove. It only works if the user was the one to add it to the table
Added a test method to print all the expenses in the expense table

icomp4 commented on src/main/java/org/compi/csc311group3/database/DbConnection.java:
```diff
@@ -37,4 +37,37 @@ password VARCHAR(200)
+     |  5 |  75.00 | Food        | Grocery shopping        | 2024-11-17 14:30:45 |       2 |
+     +----+--------+-------------+-------------------------+---------------------+---------+
+*/
+     public void initializeExpenseTable() {
```
I believe this should be within the getConnection() method. Perhaps we create Create[name} methods for each table, that we call inside getConnection. Do you have anything against this? I can implement it quickly and commit into this PR if you'd like me to. It would negate the need to call initializeDatabase in other classes.

matthewrockwood replied:
yea go ahead, i dont mind at all

icomp4 committed:
- Created an initializeTables method within DbConnection
- initializeTables can easily be expanded when we add more tables
- Removed initializeExpenseTable and initializeDatabase
